File: LCCI/1-1.is-unique-lcci/is-unique-lcci.go
func isUnique(astr string) bool {
    if len(astr) > 128 {
        return false
    }

    var char_set = make(map[byte]bool)
    for i:=0;i<len(astr);i++ {
        if (char_set[astr[i]]) {
            return false
        }
        char_set[astr[i]] = true
    }
    return true
}

File: LCCI/1-1.is-unique-lcci/is-unique-lcci.java
class Solution {
    public boolean isUnique(String astr) {
        if (astr.length() > 128) return false;

        boolean[] char_set = new boolean[128];
        for (int i = 0;i < astr.length();i++) {
            int val = astr.charAt(i);
            if (char_set[val]) {
                return false;
            }
            char_set[val] = true;
        }
        return true;
    }
}
